### PR TITLE
Use default value when index.number_of_replicas is null

### PR DESCRIPTION
**Pull request by chishui**
### Description
When creating index, if index.number_of_replicas is explicitly set to null, will use default value.

### Related Issues
Resolves https://github.com/opensearch-project/OpenSearch/issues/14783

### Check List
- [x] Functionality includes testing.
- [ ] API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md), if applicable.
- [ ] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose), if applicable.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: server/src/main/java/org/opensearch/cluster/metadata/MetadataCreateIndexService.java [hunk 1]
```diff
@@ -946,7 +946,8 @@ static Settings aggregateIndexSettings(
         if (INDEX_NUMBER_OF_SHARDS_SETTING.exists(indexSettingsBuilder) == false) {
             indexSettingsBuilder.put(SETTING_NUMBER_OF_SHARDS, INDEX_NUMBER_OF_SHARDS_SETTING.get(settings));
         }
-        if (INDEX_NUMBER_OF_REPLICAS_SETTING.exists(indexSettingsBuilder) == false) {
+        if (INDEX_NUMBER_OF_REPLICAS_SETTING.exists(indexSettingsBuilder) == false
+            || indexSettingsBuilder.get(SETTING_NUMBER_OF_REPLICAS) == null) {
```
Comment (vikasvb90) [inline server/src/main/java/org/opensearch/cluster/metadata/MetadataCreateIndexService.java:950]: Is this scenario applicable to REST layer as well or it is only applicable while using client?

**Review by vikasvb90 — APPROVED** [review]

File: server/src/main/java/org/opensearch/cluster/metadata/MetadataCreateIndexService.java (hunk 1 above)
Comment (chishui) [reply server/src/main/java/org/opensearch/cluster/metadata/MetadataCreateIndexService.java:950]: I tested through making PUT /index API, and it fell into this path

Comment (ashking94) [inline server/src/main/java/org/opensearch/cluster/metadata/MetadataCreateIndexService.java:950]: Based on the issue you shared, there is also another way in which you can get the intended behaviour. The below code that result in NPE can be revised to the following change -
https://github.com/opensearch-project/OpenSearch/blob/e288962d2870efa28b3dc67f5bcb7ad1a2f9d57f/server/src/main/java/org/opensearch/index/IndexSettings.java#L1262-L1264

```
    public int getNumberOfReplicas() {
        return IndexMetadata.INDEX_NUMBER_OF_REPLICAS_SETTING.get(settings);
    }
```

Can we also evaluate this option?

**Review by ashking94 — COMMENTED** [review]
Let's also add ITs (Integration tests). You can refer https://github.com/opensearch-project/OpenSearch/blob/main/server/src/internalClusterTest/java/org/opensearch/action/admin/indices/create/CreateIndexIT.java.

File: server/src/main/java/org/opensearch/cluster/metadata/MetadataCreateIndexService.java (hunk 1 above)
Comment (chishui) [reply server/src/main/java/org/opensearch/cluster/metadata/MetadataCreateIndexService.java:950]: Suggested code will get default value from [here](https://github.com/opensearch-project/OpenSearch/blob/9a876245f01f15cf58d3587f6b651e8181658a2b/server/src/main/java/org/opensearch/cluster/metadata/IndexMetadata.java#L239) which is always 1 regardless of cluster setting.

**Comment by chishui** [comment]
> Let's also add ITs (Integration tests). You can refer https://github.com/opensearch-project/OpenSearch/blob/main/server/src/internalClusterTest/java/org/opensearch/action/admin/indices/create/CreateIndexIT.java.

IT added

**Comment by chishui** [comment]
@vikasvb90 @ashking94 could you help add a backport to 2.x label?

**Review by dblock — APPROVED** [review]

**Pull request merged**